pdf_to_image.module

```diff
@@ -195,7 +213,17 @@ function pdf_to_image_entity_presave($entity, $type) {
       // Get the temporary and final destinations for this image.
       $filename = pathinfo($file->uri, PATHINFO_FILENAME) . '-' . $page . '.' . $settings['format'];
       $temp_destination = file_directory_temp() . '/' . $filename;
-      $final_destination = file_field_widget_uri(field_info_field($file_field['field_name']), $file_field) . $filename;
+
+      // use the configured directory of the imagefield as destination
+      $image_field_instance = field_info_instance($entity_type, $image_field, $bundle);
+      $image_dir_uri = file_field_widget_uri($image_field_info, $image_field_instance);
+      // create directory on the first run
+      file_prepare_directory($image_dir_uri, FILE_CREATE_DIRECTORY);
+      // avoid adding a third slash if it's the files root directory identified by 'public://'
+      if(substr($image_dir_uri, -1) != '/') {
+        $image_dir_uri .= '/';
+      }
+      $final_destination = $image_dir_uri . $filename;
```

Comment:
Let's also split this off into a separate issue, as this'll change current functionality and may need to take existing sites into consideration.

Reply:
Well, isn't this "new" functionality just the expected behavior? Should a configured file directory not always be honored?
And site owners who didn't care, where the images files are stored - i.e. how didn't configure a directory will not experience any change - at all!

Reply:
but I can humor you here as well - thank you for taking the time to review this!